rules/models/changes/mapping.go

```diff
@@ -0,0 +1,60 @@
+package changes
+
+import "github.com/m3db/m3metrics/rules/models"
+
+// MappingRuleChange is a mapping rule diff.
+type MappingRuleChange struct {
+	Op       UpdateOp            `json:"updateOp"`
+	RuleID   string              `json:"ruleID,omitempty"`
+	RuleData *models.MappingRule `json:"ruleData,omitempty"`
+}
+
+type mappingRuleChangesByOpNameAsc []MappingRuleChange
```

Comment:
Can just call this `mappingRuleChangesByOpAscNameAscIDAsc`, then do
```
func (a mappingRuleChangesByOpAscNameAscIDAsc) Len() int      { return len(a) }
func (a mappingRuleChangesByOpAscNameAscIDAsc) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
func (a mappingRuleChangesByOpAscNameAscIDAsc) Less(i, j int) bool {
  if a[i].Op < a[j].Op {
    return true
  }
  if a[i].Op > a[j].Op {
    return false
  }
  // For adds and changes.
  if a[i].RuleData != nil && b[i].RuleData != nil {
    return a[i].RuleData.Name < b[i].RuleData.Name
  }
  // For deletes.
  if a[i].RuleID != nil && b[i].RuleID != nil {
    return *a[i].RuleID < *b[i].RuleID
  }
  // We should not get here.
  return false
}
```

Similar for sorting rollup rule changes.